### PR TITLE
Cross client work 

### PR DESCRIPTION
**Pull request by bobbyrullo**
work for #371 

With this commit dex will perform cross-client auth for trusted peers. 

Refresh tokens are not handled yet, nor is updating trusted peers through worker api.

### CODE REVIEW

**Comment by bobbyrullo** [comment]
- [x] Add new table, field
- [ ] add API
  - [ ] worker API
  - [ ] admin API
- [ ] OIDC Work
- [ ] Scope UI

File: server/server.go [hunk 1]
```diff
@@ -39,21 +39,37 @@ const (
...
+
 	// RefreshToken takes a previously generated refresh token and returns a new ID token
 	// if the token is valid.
 	RefreshToken(creds oidc.ClientCredentials, token string) (*jose.JWT, error)
+
 	KillSession(string) error
 }
 
+// DexServer is an OIDCServer that also has dex-specific features.
+type DexServer interface {
```
Comment (ericchiang) [inline server/server.go]: Why a new interface? When would we use OIDCServer in dex over DexServer?

Comment (bobbyrullo) [inline server/server.go]: Mostly for pedantic reasons - This CrossClient auth stuff is not vanilla OIDC. If we ever pulled out the OIDC server interface, users wouldn't have to implement weird dex features. If we had a test OIDCServer it wouldn't have to implement DexServer stuff for things that don' t need it.

Happy to take it out if you think it's unnecessary..

Comment (ericchiang) [inline server/server.go]: I think the OIDC server already has very dex specific assumptions. For example it assumes that refresh tokens will be JWT and, in addition, doesn't provide a way to interpret them so it's unclear how we'd use this interface as a programmable entry point in the future.

I don't like introducing more interfaces we have exactly one implementation of.

**Comment by ericchiang** [comment]
Other than our comment thread lgtm.

File: server/server.go (hunk 1 above)
Comment (bobbyrullo) [inline server/server.go]: btw - refresh tokens are *not* JWTs - that's the ID token that gets returned when refreshing using a refresh token.

**Pull request merged**